packages/remark-math/readme.md

```diff
@@ -80,38 +125,94 @@ The default export is `remarkMath`.
 
 ### `unified().use(remarkMath[, options])`
 
-Parse and stringify math.
-
-Get’s useful when combined with [`rehype-katex`][rehype-katex] or
-[`rehype-mathjax`][rehype-mathjax].
-
-See [`micromark/micromark-extension-math`][extension-math] for more info on what
-syntax is supported.
+Plugin to support math.
 
 ##### `options`
 
+Configuration (optional).
+
 ###### `options.singleDollarTextMath`
 
 Whether to support math (text) with a single dollar (`boolean`, default:
 `true`).
 Single dollars work in Pandoc and many other places, but often interfere with
 “normal” dollars in text.
 
-#### Notes
+## Syntax
+
+This plugin applies a micromark extensions to parse the syntax.
+See its readme for parse details:
+
+*   [`micromark-extension-math`](https://github.com/micromark/micromark-extension-math#syntax)
+
+> 👉 **Note**: `$math$` works similar to `` `code` ``.
+> That means escapes don’t work inside math but you can use more dollars around
+> the math instead: `$$\raisebox{0.25em}{$\frac a b$}$$`
 
-##### Escaping
+## HTML
 
-Markdown escapes don’t work in math, similar to code, but you can use more
-dollars around the math instead: `$$\raisebox{0.25em}{$\frac
-a b$}$$`
+This plugin integrates with [`remark-rehype`][remark-rehype].
+When mdast (markdown AST) is turned into hast (the HTML AST) the math nodes
+are turned into `<span class=math-inline>` and `<div class=math-block>`
+elements.
+
+## Syntax tree
+
+This plugin applies one mdast utility to build and serialize the AST.
+See its readme for the node types supported in the tree:
+
+*   [`mdast-util-math`](https://github.com/syntax-tree/mdast-util-math#syntax-tree)
+
+## Types
+
+This package is fully typed with [TypeScript][].
+It exports an extra `Options` type which models the interface of the accepted
+options.
+
+If you’re working with the syntax tree, make sure to import this plugin
+somewhere in your types, as that registers the new node types in the tree.
+
+```js
+/** @typedef {import('remark-math')} */
```

Comment:
:thinking: `remark-math` is probably already included at the top level?
Wouldn't that include the additional node types?
Or is this specifically for plugins which may not have a root unified pipeline, but still depend on the node types from math?

Reply:
yeah the last part.

This isn’t a thing for GFM, or mostly for frontmatter, because those are in `@types/mdast`. But for directives (https://github.com/remarkjs/remark-directive#types) and math, something needs to be done